st2common/st2common/models/api/keyvalue.py

```diff
@@ -145,16 +160,24 @@ def to_model(cls, kvp):
         else:
             expire_timestamp = None
 
-        if getattr(kvp, 'secret', False):
+        secret = getattr(kvp, 'secret', False)
+
+        if secret:
             if not KeyValuePairAPI.crypto_key:
                 msg = ('Crypto key not found in %s. Unable to encrypt value for key %s.' %
                        (KeyValuePairAPI.crypto_key_path, name))
                 raise CryptoKeyNotSetupException(msg)
             value = symmetric_encrypt(KeyValuePairAPI.crypto_key, value)
-            secret = True
 
-        model = cls.model(name=name, description=description, value=value,
-                          secret=secret,
+        scope = getattr(kvp, 'scope', SYSTEM_SCOPE)
+
+        if scope not in ALLOWED_SCOPES:
```

Comment:
Is this really the right place for that?

Imo, it kinda belongs in the API controller and we should return 400 if user provides an invalid scope...